File: server/src/protocol/StopStreamCommand.cpp
#include "StopStreamCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

StopStreamCommand::StopStreamCommand()
        : Command(10, CommandType::STOP_STREAM)
{}

StopStreamCommand::~StopStreamCommand()
{}

vector<uint8_t> StopStreamCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string StopStreamCommand::getName()
{
    return string("StopStreamCommand");
}

void StopStreamCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void StopStreamCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/StopStreamCommand.java
package wpam.mobile_client.protocol;
import java.util.ArrayList;

public final class StopStreamCommand extends Command
	{
		public StopStreamCommand()
		{
			super(10, CommandType.STOP_STREAM);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public  ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("StopStreamCommand");
		}
		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}